Model/Api/DiscountCodeValidation.php

```diff
@@ -282,6 +282,12 @@ public function validate()
                 $giftCard = $this->discountHelper->loadAmastyGiftCard($couponCode);
             }
 
+            // Apply Mageplaza_GiftCard
+            if (empty($giftCard)) {
+                // Load the gift card by code
+                $giftCard = $this->discountHelper->loadMageplazaGiftCard($couponCode);
+            }
+
```

Comment:
You should add a section below in applyingGiftCardCode method where this $giftCard is applied to quote / immutable quote. Every gift cart module has it's own logic.

Reply:
I've updated it.